### PR TITLE
New Buildkite instrumentation format

### PR DESCRIPTION
**Pull request by paulbalaji**
Added `traceStart` and `traceEnd` functions to pinned tools. Similar use to pushd/popd, but for marking trace spans

Drive-by reindented several bash files to a consistent 4 spaces

### CODE REVIEW

File: scripts/pinned-tools.sh [hunk 1]
```diff
@@ -1,89 +1,102 @@
...
 
 function cleanUnity() {
-  rm -rf "${1}/Library"
-  rm -rf "${1}/Temp"
+    rm -rf "${1}/Library"
+    rm -rf "${1}/Temp"
 }
 
-echo "--- Sourcing pinned tools :round_pushpin:"
+function traceStart() {
```
Comment (zeroZshadow) [inline scripts/pinned-tools.sh:25]: Could this somehow reject / strip traces that contain a `;`? To avoid issues

**Review by zeroZshadow — APPROVED** [review]

File: scripts/pinned-tools.sh (hunk 1 above)
Comment (paulbalaji) [reply scripts/pinned-tools.sh:25]: ye

File: ci/docker/entrypoint.sh [hunk 2]
```diff
@@ -18,11 +18,16 @@ chown -R user:user "$(pwd)/tools"
...
-gosu user dotnet test --logger:"nunit;LogFilePath=${TEST_RESULTS_DIR}/docslinter-test-results.xml" "tools/DocsLinter/DocsLinter.csproj"
+traceStart "Build Tools.sln :construction:"
+    gosu user dotnet build tools/Tools.sln
+traceEnd
 
-echo "--- Test ReleaseTool.csproj :fork:"
-gosu user dotnet test --logger:"nunit;LogFilePath=${TEST_RESULTS_DIR}/releasetool-test-results.xml" "tools/ReleaseTool.Tests/ReleaseTool.Tests.csproj"
+traceStart "Test DocsLinter.csproj :link:"
+    gosu user dotnet test --logger:"nunit;LogFilePath=${TEST_RESULTS_DIR}/docslinter-test-results.xml" "tools/DocsLinter/DocsLinter.csproj"
+traceEnd
```
Comment (jamiebrynes7) [inline ci/docker/entrypoint.sh:29]: Do we use this anymore?

File: scripts/pinned-tools.sh [hunk 3]
```diff
@@ -1,89 +1,102 @@
...
-    export PATH="${PATH}:/usr/local/share/dotnet/"
-  fi
-fi
+traceStart "Sourcing pinned tools :round_pushpin:"
+    # Ensure for the Mac TC agents that dotnet is on the path.
+    if isMacOS; then
+        if ! which dotnet; then
+            export PATH="${PATH}:/usr/local/share/dotnet/"
+        fi
+    fi
```
Comment (jamiebrynes7) [inline scripts/pinned-tools.sh:43]: Wonder if this is still required for the BK agents

File: scripts/pinned-tools.sh [hunk 4]
```diff
@@ -1,89 +1,102 @@
...
+        BRANCH="${BUILDKITE_BRANCH}"
+    else
+        BRANCH=$(git branch | sed -n -e 's/^\* \(.*\)/\1/p')
+    fi
+
+    if [[ "${BRANCH}" == docs/* ]]; then
+        return 0
+    fi
+    return 1
 }
```
Comment (jamiebrynes7) [inline scripts/pinned-tools.sh]: Similar comment - is this still used?

Comment (paulbalaji) [reply scripts/pinned-tools.sh]: nop, will delete

File: ci/docker/entrypoint.sh (hunk 2 above)
Comment (paulbalaji) [reply ci/docker/entrypoint.sh:29]: We don't :'(

I'll make a ticket to delete it + the project because I don't want to bloat this PR

Comment (paulbalaji) [reply ci/docker/entrypoint.sh:29]: https://improbableio.atlassian.net/browse/UTY-2620

**Review by jamiebrynes7 — APPROVED** [review]
LGTM

**Pull request merged**